### DISCUSSION TITLE
Youtube Analytics features as exposed by Youtube studio

### DISCUSSION THREAD

**Discussion by Proful** [LuanRT/YouTube.js#24] [General]
As a youtube creator point of view, can we have api exposed to fetch

- video details
- subscriber gain per each video
- watch time etc

**Comment by LuanRT** [comment]
Totally possible, I've been playing around with some endpoints related to YouTube Studio and they seem to be pretty straightforward. So I'll probably implement it as soon as I have more time to further investigate this.

**Reply by Proful** [reply]
Thanks a lot!

**Reply by rakinne** [reply]
Just came across the library yesterday - more research needed but it looks robust. Each of the concerns Proful highlighted could be made into issues

**Comment by seomikewaltman** [comment]
In order to get Studio analytic's /join endpoints working you'll need to crack the salted sapisidhash. I'd caution against exposing that in an open repository. YouTube watches these type of repos for vulnerabilities and if they see their hash algo exposed they'll start implementing recaptcha on more of their Studio endpoints like they have on /metadata_update. TubeBuddy, VidIQ bulk features will get killed and that would be a detriment to the creator community as a whole. Everything @Proful requested can be pulled from the YouTube Analytics API for free using their Nodejs SDK.

**Reply by LuanRT** [reply]
> In order to get Studio analytic's /join endpoints working you'll need to crack the salted sapisidhash. I'd caution against exposing that in an open repository. YouTube watches these type of repos for vulnerabilities and if they see their hash algo exposed they'll start implementing recaptcha on more of their Studio endpoints like they have on /metadata_update. TubeBuddy, VidIQ bulk features will get killed and that would be a detriment to the creator community as a whole. Everything @Proful requested can be pulled from the YouTube Analytics API for free using their Nodejs SDK.

I see what you're saying, but as far as I can tell the sapisidhash is not really a complicated algorithm and is only required when you sign in using cookies, I'd argue they're more interested in protecting their video CDNs. 

Also, we'll be mostly using the Android Studio app endpoints which do not have any sort of anti bot protection.

**Reply by seomikewaltman** [reply]
Nice. I'll hope to contribute as much as I can. I've spent 2 years doing this very thing for my Notifications for YouTube chrome extension and a few advanced Studio analytics extensions to get at the CTRs and Video impressions :)

**Reply by pmkemper** [reply]
Did you make progress on making the analytics' /join endpoints work?
